dkist/io/asdf/converters/tiled_dataset.py

```diff
@@ -33,7 +33,7 @@ def to_yaml_tree(cls, tiled_dataset, tag, ctx):
         meta.pop("history", None)
         tree["meta"] = meta
         # Copy the data as well so we aren't editing dataset headers in place
-        datasets = copy.copy(tiled_dataset._data)
+        datasets = copy.deepcopy(tiled_dataset._data)
```

Comment:
oh really? This makes me worry it's going to copy more stuff than I expected and take more memory.

Reply:
Don't disagree but just `copy` doesn't copy enough stuff it seems :shrug:

Reply:
What doesn't it copy?!

Reply:
oh sure, copy would only copy the array and not the references to the dataset's within.

We want the copy to copy the dataset objects but not the stuff contained within them :thinking:

Reply:
Indeed. Corresponding commits incoming.